### PR TITLE
Manage translations through react-intl

### PR DESCRIPTION
**Pull request by mbenadda**
## Purpose

This is step 3 & 4 of #89. We need to manage strings inside the JS app and to collect them for translation.

## Proposal

- [x] Add a dependency (`react-intl`) to manage i18n for us
- [x] Mark all strings and replace them with full message descriptors
- [x] Automatically extract them during `yarn build` through a babel plugin

### CODE REVIEW

File: src/richie-front/js/components/CourseGlimpse/CourseGlimpse.spec.tsx [hunk 1]
```diff
@@ -18,12 +19,14 @@ describe('components/CourseGlimpse', () => {
...
-      <CourseGlimpse course={course} organizationMain={organization} />,
+      <IntlProvider>
+        <CourseGlimpse course={course} organizationMain={organization} />
+      </IntlProvider>,
     );
 
     expect(wrapper.html()).toContain('Course 42');
     expect(wrapper.find('img').attr('src')).toContain('/thumbs/small.png');
-    expect(wrapper.html()).toContain('Starts on Mar 12, 2018');
+    expect(wrapper.html()).toContain('Starts on', 'Mar 12, 2018');
```
Comment (sampaccoud) [inline src/richie-front/js/components/CourseGlimpse/CourseGlimpse.spec.tsx]: Curious: why this change?

File: tsconfig.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "compilerOptions": {
     "allowSyntheticDefaultImports": true,
     "esModuleInterop": true,
-    "jsx": "react",
+    "jsx": "preserve",
```
Comment (sampaccoud) [inline tsconfig.json]: curious why this change?

File: src/richie-front/js/components/CourseGlimpse/CourseGlimpse.spec.tsx (hunk 1 above)
Comment (mbenadda) [reply src/richie-front/js/components/CourseGlimpse/CourseGlimpse.spec.tsx]: There is now a `<span></span>` around the date

File: tsconfig.json (hunk 2 above)
Comment (mbenadda) [reply tsconfig.json]: See my note in the 2nd commit message:
> Note: we now need TypeScript to leave JSX in its output so it can be picked up by the babel plugin. [...]

File: src/richie-front/js/components/CourseGlimpse/CourseGlimpse.spec.tsx (hunk 1 above)
Comment (sampaccoud) [reply src/richie-front/js/components/CourseGlimpse/CourseGlimpse.spec.tsx]: oh then maybe it's a good idea to include the span in the test?
